aim/sdk/run.py

```diff
@@ -454,7 +454,8 @@ def _track_impl(
     @property
     def props(self):
         if self._props is None:
-            self._props = self.repo.request_props(self.hash, self.read_only)
+            with self.repo.lock():
```

Comment:
check if the `Repo` is for remote tracking. no need to create a lock for that case.